### DISCUSSION TITLE
TuxRacer port

### DISCUSSION THREAD

**Discussion by oherau** [OnionUI/Onion#1555] [Ideas]
Hi,

Based on the toolchain used for the build of the "Battle for Wesnoth" port, I tried to build a port of TuxRacer.
I modified the CMakeLists.txt to use CMAKE 3.13 -the highest version compatible with the toolchain-
I almost succeded. Compiling is ok but the link fails and displays the following message :
`/usr/bin/ld: cannot find -lSDL
/usr/bin/ld: cannot find -lSDL_image
/usr/bin/ld: cannot find -lSDL_mixer
/usr/bin/ld: cannot find -lfreetype
/usr/bin/ld: cannot find -lGL
/usr/bin/ld: cannot find -lGLU
collect2: error: ld returned 1 exit status`

I think the script is missing some fine tuning (-L<path_to_lib>) to work properly for this game but I couldn't make it.

Questions I'm asking to myself:
1-are all the packages usefull ? (maybe not)
2-is the Miyoo able to handle OpenGL ? (hardware or software rendering)
3-does the Miyoo match minimum requirements for this games (I could not find the info)

Here is the script I used with the toolchain:

**mk_tuxracer.sh:**
[mk_tuxracer.sh.txt](https://github.com/OnionUI/Onion/files/14967925/mk_tuxracer.sh.txt)

**Comment by XK9274** [comment]
There's no configure in this project, it may ignore most of the exports aswell. So you'll have to modify the Makefile.

Just to add to this, do those libraries exist in your environment and are they linked correctly with LDFLAGS, and the includes with CXX flags?

**Comment by oherau** [comment]
this line was missing in the mk_tuxracer.sh script:
`export CXX=${CROSS_COMPILE}-g++`

I also modified the Makefile to add (generic linux section):
CXXFLAGS = -Wall -Wextra -O2 -DOS_LINUX -I/usr/include/freetype2 **-I/usr/include**
LDFLAGS  = -lSDL -lSDL_image -lSDL_mixer -lfreetype **-L/usr/lib/arm-linux-gnueabihf**

I still have some errors at link but it's better
